google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/standard/AppEngineStandardWebIntegration.java

```diff
@@ -42,21 +44,35 @@
 import com.intellij.remoteServer.impl.configuration.deployment.DeployToServerConfigurationType;
 import com.intellij.remoteServer.impl.configuration.deployment.DeployToServerConfigurationTypesRegistrar;
 import com.intellij.remoteServer.impl.configuration.deployment.DeployToServerRunConfiguration;
+import com.intellij.util.ArrayUtil;
 import com.intellij.util.containers.ContainerUtil;
+import java.io.IOException;
 import java.util.List;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 /** @author nik */
 public abstract class AppEngineStandardWebIntegration {
+  private static final Logger LOG = Logger.getInstance(AppEngineStandardWebIntegration.class);
 
   public static AppEngineStandardWebIntegration getInstance() {
     return ServiceManager.getService(AppEngineStandardWebIntegration.class);
   }
 
   @Nullable
-  public abstract VirtualFile suggestParentDirectoryForAppEngineWebXml(
-      @NotNull Module module, @NotNull ModifiableRootModel rootModel);
+  public VirtualFile suggestParentDirectoryForAppEngineWebXml(
+      @NotNull Module module, @NotNull ModifiableRootModel rootModel) {
+    final VirtualFile root = ArrayUtil.getFirstElement(rootModel.getContentRoots());
```

Comment:
Looks like `getContentRoots()` returns a collection, would it be enough in all cases to just work with the first content root?

Reply:
Also here, just for clarification on IDE model here, I think there could be source and resources content roots, are they all returned here or it's only source ones?